File: LogisimPlus/LogisimPlus/logisim/std/memory/JKFlipFlop.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.std.memory
{
	using Value = logisim.data.Value;

	public class JKFlipFlop : AbstractFlipFlop
	{
		public JKFlipFlop() : base("J-K Flip-Flop", "jkFlipFlop.gif", Strings.getter("jkFlipFlopComponent"), 2, false)
		{
		}

		protected internal override string getInputName(int index)
		{
			return index == 0 ? "J" : "K";
		}

		protected internal override Value computeValue(Value[] inputs, Value curValue)
		{
			if (inputs[0] == Value.FALSE)
			{
				if (inputs[1] == Value.FALSE)
				{
					return curValue;
				}
				else if (inputs[1] == Value.TRUE)
				{
					return Value.FALSE;
				}
			}
			else if (inputs[0] == Value.TRUE)
			{
				if (inputs[1] == Value.FALSE)
				{
					return Value.TRUE;
				}
				else if (inputs[1] == Value.TRUE)
				{
					return curValue.not();
				}
			}
			return Value.UNKNOWN;
		}
	}

}

File: logisim/app/src/main/java/logisim/std/memory/JKFlipFlop.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.std.memory;


import logisim.data.WireValue.WireValue;
import logisim.data.WireValue.WireValues;

public class JKFlipFlop extends AbstractFlipFlop {
	public JKFlipFlop() {
		super("J-K Flip-Flop", "jkFlipFlop.gif", Strings.getter("jkFlipFlopComponent"), 2, false);
	}

	@Override
	protected String getInputName(int index) {
		return index == 0 ? "J" : "K";
	}

	@Override
	protected WireValue computeValue(WireValue[] inputs, WireValue curValue) {
		if (inputs[0] == WireValues.FALSE) {
			if (inputs[1] == WireValues.FALSE) return curValue;
			else if (inputs[1] == WireValues.TRUE) return WireValues.FALSE;
		} else if (inputs[0] == WireValues.TRUE) if (inputs[1] == WireValues.FALSE) return WireValues.TRUE;
		else if (inputs[1] == WireValues.TRUE) return curValue.not();
		return WireValues.UNKNOWN;
	}
}
